miden-proto/proto/rpc.proto

```diff
@@ -0,0 +1,74 @@
+// Specification of the user facing gRPC API.
+//
+// This server is a thin layer over the other services, it will mostly forward
+// requests downstream.
+syntax = "proto3";
+package rpc;
+
+import "account.proto";
+import "block.proto";
+import "digest.proto";
+import "note.proto";
+import "transaction.proto";
+
+message GetBlockHeaderRequest {
+    block.BlockSpec block_spec = 1;
+}
+
+message GetNoteRequest {
+    note.NoteSpec note_spec = 2;
+}
+
+message GetTransactionDataRequest {
+    account.AccountId account_id = 1;
+    block.BlockSpec block_spec = 2;
+    repeated note.NoteOrigin notes = 3;
+}
+
+message ListNotesByTagRequest {
+    // Only include tags after this block.
+    block.BlockSpec block_spec = 1;
+    // Minimum value of the tag.
+    fixed64 tag_min = 2;
+    // Maximum value of the tag.
+    fixed64 tag_max = 3;
+}
+
+message SubmitTransactionRequest {
+    transaction.ProvenTransaction transaction = 1;
+}
+
+// Server response. The format of the data depends on the endpoint, it is the
+// responsability of the client to decode the data form its original format.
+message ResponseBytes {
+    bytes data = 1;
+}
+
+service Api {
+    // QUERY ENPOINTS
+    // -------------------------------------------------------------------------------------------
+
+    // Query a block header by block spec.
+    rpc GetBlockHeader(GetBlockHeaderRequest) returns (ResponseBytes) {}
+
+    // Query a note by note spec.
+    rpc GetNote(GetNoteRequest) returns (ResponseBytes) {}
+
+    // Query for nullifiers.
+    rpc CheckNullifiers(stream digest.Digest) returns (stream ResponseBytes) {}
+
+    // Query data required for transaction execution.
+    rpc GetTransactionData(GetTransactionDataRequest) returns (ResponseBytes) {}
```

Comment:
This endpoint would be used by internal components and thus does not need to be exposed via public RPC.

Reply:
Getting transaction data is super useful for debugging, specially for network accounts.

Should I remove it?

Reply:
removed

Reply:
> Getting transaction data is super useful for debugging, specially for network accounts.

Yes, but we still need to figure out the exact design and there probably will be several endpoints for getting transaction data. For example, `GetTransactionData` will probably be an internal endpoint which retrieves the data required for executing a given transaction.